### DISCUSSION TITLE
Setting head meta in script setup syntax

### DISCUSSION THREAD

**Discussion by iamschulz** [nuxt/nuxt#18144] [Questions]
Hi!
I'm trying to modify the `<title>` inside a page component, while simultaneously using the `<script setup>` syntax. I know I can do something like
```typescript
<script lang="ts">
export default {
	head() {
		return {
			title: 'New title',
		};
	},
};
</script>
<script setup lang="ts">
 // ...
</script>
```

But now I'd like to add route information to the title:

```typescript
<script lang="ts">
export default {
	head() {
		const route = useRoute();

		return {
			title: `${route.params.project} | MySite`,
		};
	},
};
</script>

<script setup lang="ts">
import { useRoute } from "vue-router";
const route = useRoute();
</script>
```

This works fine inside the dev server, but breaks when deploying a prod build to Netlify:
```
Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'params')
```

Is there a way I can use `head()` inside `<script setup>`, where I already have access to `route`? Where would I import it from? That would solve the described problem as well as get rid of the combined `script setup` and `export default` syntax, which seems awkward to me.

**Comment by viii47** [comment]
Hey @iamschulz check https://v3.nuxtjs.org/docs/usage/meta-tags#meta-components 

You can either directly inject your head into the template section or with `useMeta({ ... })` inside the setup section.
